src/tinuous/util.py

```diff
@@ -46,17 +47,78 @@ def iterfiles(dirpath: Path) -> Iterator[Path]:
                 yield p
 
 
+class LazySlicingFormatter(Formatter):
+    def __init__(self, var_defs: Dict[str, str]):
+        self.var_defs: Dict[str, str] = var_defs
+        self.expanded_vars: Dict[str, str] = {}
+        super().__init__()
+
+    def get_value(
+        self, key: Union[int, str], args: Sequence[Any], kwargs: Mapping[str, Any]
+    ) -> Any:
+        if isinstance(key, int):
+            return args[key]
+        elif key in kwargs:
+            return kwargs[key]
+        elif key in self.expanded_vars:
+            return self.expanded_vars[key]
```

Comment:
oh -- that seems like quite an elaborate (smells like a more proper of cause) solution! ;)   Since would be in the "core" of things, could you add unittests for not yet covered cases in this function?